internal/rcm/api.go

```diff
@@ -0,0 +1,148 @@
+// Package rcm provides alternative HTTP API to Weldr.
+// It's primary use case is for the RCM team. As such it is driven solely by their requirements.
+package rcm
+
+import (
+	"encoding/json"
+	"log"
+	"net"
+	"net/http"
+
+	"github.com/google/uuid"
+	"github.com/julienschmidt/httprouter"
+	"github.com/osbuild/osbuild-composer/internal/blueprint"
+	"github.com/osbuild/osbuild-composer/internal/store"
+	"github.com/osbuild/osbuild-composer/internal/target"
+)
+
+// API encapsulates RCM-specific API that is exposed over a separate TCP socket
+type API struct {
+	logger *log.Logger
+	store  *store.Store
+	router *httprouter.Router
+}
+
+// New creates new RCM API
+func New(logger *log.Logger, store *store.Store) *API {
+	api := &API{
+		logger: logger,
+		store:  store,
+		router: httprouter.New(),
+	}
+
+	api.router.RedirectTrailingSlash = false
+	api.router.RedirectFixedPath = false
+	api.router.MethodNotAllowed = http.HandlerFunc(methodNotAllowedHandler)
+	api.router.NotFound = http.HandlerFunc(notFoundHandler)
+
+	api.router.POST("/submit", api.submit)
+	api.router.GET("/status/:uuid", api.status)
```

Comment:
Do we want versioning? `GET /v1/compose/:uuid`?

Reply:
I don't have any strong opinion about this. I decided not to include it yet because I'm not sure it is necessary, but I'm fine with including it.

Reply:
I don't have any strong opinion about this. I decided not to include it yet because I'm not sure it is necessary, but I'm fine with including it.

Reply:
I don't have it neither. Better introduce it sooner than later?

Reply:
I think I agree. The alternative would be to introduce a new socket/listening port if ever we wanted a second version of the API. I think I prefer the option of just changing the version in the endpoint, but I think @larskarlitski also had some opinions on this when I introduced the jobqueue API (we should probably do the same in both, but happy to change the jobqueue if we agree that that's not right).